test/e2e/certificates_test.go

```diff
@@ -267,6 +270,84 @@ var _ = Describe("ACME Certificate", Ordered, func() {
 			err = verifyCertificate(ctx, certName, ns.Name, certDomain)
 			Expect(err).NotTo(HaveOccurred())
 		})
+
+		It("should obtain a valid LetsEncrypt certificate in GCP mint/passthrough clusters-65035", func() {
+
+			By("Getting Infrastructure object")
+			infra, err := configClient.Infrastructures().Get(ctx, "cluster", metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			if infra.Status.PlatformStatus.Type != configv1.GCPPlatformType {
+				Skip("Skip this case for current plaform, it's not gcp.")
+			}
+
+			By("Check cloud credential in cluster")
+			_, err = loader.KubeClient.CoreV1().Secrets("kube-system").Get(ctx, "gcp-credentials", metav1.GetOptions{})
+			if err != nil {
+				Skip("Skipping for the cluster without credential in cluster")
+			}
+
+			By("Creating CredentialsRequest object")
+			loader.CreateFromFile(testassets.ReadFile, filepath.Join("testdata", "credentials", "credentialsrequest_gcp.yaml"), "")
+
+			By("Waiting for cloud secret to be available")
+			// The name is defined cloud credential by the testdata YAML file.
+			credentialSecret := "gcp-credentials"
+			err = wait.PollImmediate(PollInterval, TestTimeout, func() (bool, error) {
+				_, err := loader.KubeClient.CoreV1().Secrets("cert-manager").Get(ctx, credentialSecret, metav1.GetOptions{})
+				if err != nil {
+					return false, nil
+				}
+				return true, nil
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Configure cert-manager to use credential, setting this credential secret name in subscription object")
+			err = patchSubscriptionWithCloudCredential(ctx, loader, credentialSecret)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating new certificate ClusterIssuer")
+			// The name is defined by the testdata YAML file clusterissuer_gcp.yaml
+			clusterIssuerName := "acme-dns01-clouddns-ambient"
+			gcpProjectId := infra.Status.PlatformStatus.GCP.ProjectID
+			Expect(gcpProjectId).NotTo(Equal(""))
+			replaceStrMap := map[string]string{
+				"OPENSHIFT_QE": gcpProjectId,
+			}
+			loadFileAndReplaceStr := func(fileName string) ([]byte, error) {
+				fileContentsStr, err := replaceStrInFile(replaceStrMap, fileName)
+				return []byte(fileContentsStr), err
+			}
+			loader.CreateFromFile(loadFileAndReplaceStr, filepath.Join("testdata", "acme", "clusterissuer_gcp.yaml"), "")
+			defer certmanagerClient.CertmanagerV1().ClusterIssuers().Delete(ctx, clusterIssuerName, metav1.DeleteOptions{})
+
+			By("Creating a test namespace")
+			ns, err := loader.CreateTestingNS("e2e-acme-ambient-dns01-65035")
+			Expect(err).NotTo(HaveOccurred())
+			defer loader.DeleteTestingNS(ns.Name)
+
+			By("Creating new certificate")
+			randomString := randomStr(7)
```

Comment:
@geliu2016 to avoid failure error again like OCPQE-16688, it is better to make randomString shorter so that the whole length of `randomString.baseDomain` will be shorter.
```suggestion
			randomString := randomStr(3)
```

Reply:
@xingxingxia ARE YOU SURE? the length of random_str is shorter at first, and you worry about that shorter randomstr will conflict with other parts of dns_name, so I added length of random_str to low the risk of same name conflict issue.
it depend on you, if you prefer shorter, I may update it.

Reply:
Updated.